iib/workers/tasks/utils.py

```diff
@@ -141,3 +143,58 @@ def run_cmd(cmd, params=None, exc_msg=None, cmd_repr=None):
         raise IIBError(exc_msg)
 
     return response.stdout
+
+
+def request_logger(func):
+    """
+    Log messages relevant to the current request to a dedicated file.
+
+    If iib_request_logs_dir is set, a temporary log hanlder is added before the decorated
+    function is invoked. It's then removed once the decorated function completes exection.
+
+    If iib_request_logs_dir is not set, the temporary log handler will not be added.
+
+    :param function func: the function to be decorated. The function must take the request_id
+        parameter.
+    :return: the decorated function
+    :rtype: function
+    """
+    log_dir = get_worker_config().iib_request_logs_dir
+    log_level = get_worker_config().iib_request_logs_level
+    log_formatter = logging.Formatter(get_worker_config().iib_request_logs_format)
+
+    @functools.wraps(func)
+    def wrapper(*args, **kwargs):
+        request_log_handler = None
+        if log_dir:
+            request_id = _get_function_arg_value('request_id', func, args, kwargs)
+            if not request_id:
+                log.warning('Unable to get "request_id" from %s', func)
```

Comment:
Shouldn't this be an error message?

Reply:
I've decided to actually raise an error here.